src/main/java/xyz/nucleoid/plasmid/game/common/team/TeamManager.java

```diff
@@ -0,0 +1,330 @@
+    }
+
+    /**
+     * Gets set of players being part of the team
+     *
+     * @param team targeted {@link GameTeam}
+     * @return Set of Players
+     */
+    public PlayerSet getPlayers(GameTeam team) {
+        PlayerSet set = this.teamToPlayerSet.get(team);
+
+        return set == null ? PlayerSet.EMPTY : set;
+    }
+
+    /**
+     * Gets set of {@link PlayerRef} being part of the team
+     *
+     * @param team targeted {@link GameTeam}
+     * @return Set of {@link PlayerRef}
+     */
+    public Set<PlayerRef> getPlayerRefs(GameTeam team) {
+        Set<PlayerRef> set = this.teamToPlayerRefSet.get(team);
+
+        return set == null ? Collections.emptySet() : set;
+    }
+
+    public boolean getFriendlyFire(GameTeam team) {
+        return this.teamFriendlyFire.getBoolean(team);
+    }
+
+    public boolean setFriendlyFire(GameTeam team, boolean value) {
+        return this.teamFriendlyFire.put(team, value);
+    }
+
+    public AbstractTeam.CollisionRule getCollisionRule(GameTeam team) {
+        return this.getVanilla(team).getCollisionRule();
+    }
+
+    public void setCollisionRule(GameTeam team, AbstractTeam.CollisionRule rule) {
+        this.getVanilla(team).setCollisionRule(rule);
+        this.sendUpdates(team);
+    }
+
+    public AbstractTeam.VisibilityRule getNameTagVisibilityRule(GameTeam team) {
+        return this.getVanilla(team).getNameTagVisibilityRule();
+    }
+
+    public void setNameTagVisibilityRule(GameTeam team, AbstractTeam.VisibilityRule rule) {
+        this.getVanilla(team).setNameTagVisibilityRule(rule);
+        this.sendUpdates(team);
+    }
+
+    public Text applyPlayerFormatting(PlayerRef player, Text text) {
+        GameTeam team = this.playerRefToTeam.get(player);
+        if (team != null) {
+            Team vanillaTeam = this.getVanilla(team);
+            return new LiteralText("").append(vanillaTeam.getPrefix()).append(text.shallowCopy().setStyle(Style.EMPTY.withColor(team.color()))).append(vanillaTeam.getSuffix());
+        }
+        return text;
+    }
+
+    public Text getPrefix(GameTeam team) {
+        return this.getVanilla(team).getPrefix();
+    }
+
+    public void setPrefix(GameTeam team, Text value) {
+        this.getVanilla(team).setPrefix(value);
+        this.sendUpdates(team);
+    }
+
+    public Text getDisplayName(GameTeam team) {
+        return this.getVanilla(team).getDisplayName();
+    }
+
+    public void setDisplayName(GameTeam team, Text value) {
+        this.getVanilla(team).setDisplayName(value);
+        this.sendUpdates(team);
+    }
+
+    public Text getSuffix(GameTeam team) {
+        return this.getVanilla(team).getSuffix();
+    }
+
+    public void setSuffix(GameTeam team, Text value) {
+        this.getVanilla(team).setSuffix(value);
+        this.sendUpdates(team);
+    }
+
+    public GameTeam getSmallestTeam() {
+        GameTeam smallest = null;
+        int count = 9999;
```

Comment:
I'm not sure if this is the best solution here, but either way it should use `Integer.MAX_VALUE` just to be safe.